core/src/test/java/com/datastax/oss/driver/internal/core/metadata/DefaultTopologyMonitorTest.java

```diff
@@ -654,4 +794,12 @@ private void assertLog(Level level, String message) {
     assertThat(logs).hasSize(1);
     assertThat(logs.iterator().next().getFormattedMessage()).contains(message);
   }
+
+  private void assertLogContains(Level level, String message) {
+    verify(appender, atLeast(1)).doAppend(loggingEventCaptor.capture());
+    Iterable<ILoggingEvent> logs =
+        filter(loggingEventCaptor.getAllValues()).with("level", level).get();
+    assertThat(
+        Streams.stream(logs).map(ILoggingEvent::getFormattedMessage).anyMatch(message::contains));
+  }
```

Comment:
Note: this private method was added (and used in some of the logging tests above) because we (correctly) added log messages around the case of failing to detect an RPC address or port and the way the mocks are setup above will actually trigger this logic... which in turn affects the count of number of log messages in assertLog().

The right answer is probably to make the generation of mock data dependent upon the C*/DSE version you wish to mock (since system.local, system.peers and system.peers_v2 may vary between versions within both C* and DSE).  Doing so is a significant refactoring of this test, however, so for now I chose to address the problem much more simply by checking to see whether we generate the target log message _at all_ rather than whether that's the only log message we generate.